### ISSUE TITLE
Port build script to xbuild/msbuild

### ISSUE THREAD

**Issue by joephayes** [xmlunit/xmlunit.net#8]
should support IDEs as well (csproj)

**Comment by bodewig** [comment]
I've started to work on it, but this is still open for anybody else to finish.

One feature of the NAnt build that needs to be migrated to the MSBuild solution is that we create most of the assembly information from common metadata in a single place (this is the build file right now, but I could live with doing it inside a common linked C# file).  I've looked into the MSBuild extension libraries but their tasks seem to be more limited than NAnt's `asminfo` - I haven't seen a way to easily add `InternalsVisibleTo` for example.

Also, so far I haven't really looked into how far my personal "should work with xbuild" requirement is limiting us.

**Comment by bodewig** [comment]
With  ce595f4 the solution is able to build all four assemblies with the same information we get with Nant.  Still not tested with xbuild.

Things that the NAnt build does that have not been covered so far:

* run unit tests automatically
* build source/binary zips (actually I haven't adapted the NAnt build when I moved files around, so this is currently broken anyway)
* build and publish nuget packages

**Comment by bodewig** [comment]
I've got a working solution for this and #11 in a branch and am currently working on making things work on Travis.  Also I need to document how to use nuget restore (in particular on Linux).

Right now there seems to be a different interpretation of relative paths between Travis and my Linux system (using Mono 4.x on Travis and 3.x on mine) so the tests are failing there.

Unfortunately I had to fall back to `Exec` (which in turn I haven't tried on Windows) since the MSBuild.NUnit task has a hardcoded "strip the .exe suffix" logic - will open an issue over there.
